### ISSUE TITLE
Borderless window mode

### ISSUE THREAD

**Issue by legionpheonix** [joncampbell123/dosbox-x#1433]
Good afternoon.
Thank you so much for the previous help!
Tell me, can you add borderless window mode?
It would be a very useful thing.
windowresolution = desktop
Borderless = true

**Comment by tbullock** [comment]
Borderless fullscreen (or otherwise) would go a long way to improving the usability for me.  More so than pretty much any additional feature actually.

I am fairly certain that SDL supports the functionality natively:

https://wiki.libsdl.org/SDL_WindowFlags

SDL_WINDOW_FULLSCREEN_DESKTOP 
fullscreen window at the current desktop resolution

**Comment by mart0258** [comment]
Implementing this is simple. src/gui/sdlmain.cpp has function GFX_ForceFullscreenExit - when I disabled that function in DosBox-ECE, I encountered no drawbacks or unexpected behavior. Dosbox-X should be no different. 

In general, apps tend to have more problems because they handle Alt-tab by shrinking to the taskbar when they could simply ignore it and work as a window in the background - especially with modern computers where the borderless windowed impact is rather minimal.

**Comment by Wengier** [comment]
Forcing borderless window mode itself may not be difficult, but the problem is that then you cannot move the window because there will be no title bar for the move to work, and it apparently require some work to support moving the window without the title bar. So there is indeed some impact for supporting the borderless mode.

**Comment by laxdragon** [comment]
+1 For borderless fullscreen mode. For me on Windows 10 going fullscreen messes with all my window placement of my other apps. So I never use fullscreen with DOSBox-X right now.

**Comment by joncampbell123** [comment]
@laxdragon I thought I disabled the mode changing code in SDL so that DOSBox-X "fullscreen" is just a maximized window with no border, title, or menu bar. At least SDL1 builds. Are you using SDL2?

**Comment by laxdragon** [comment]
@joncampbell123 correct, I'm using the 64bit mingw SDL2 build. It tends gives me the best performance, so I default to that build.

**Comment by joncampbell123** [comment]
SDL2 builds could mirror the same behavior then if they use SDL_WINDOW_FULLSCREEN_DESKTOP instead of SDL_WINDOW_FULLSCREEN.

https://wiki.libsdl.org/SDL_SetWindowFullscreen

**Comment by RNMB15** [comment]
In the dosbox conf under "[sdl]" there is an option where you can make settings. that in full screen mode the monitor is adjusted to the original resolution of the game or application. This is what it looks like.

[sdl]
windowresolution  = original

But when I do this under dosbox-x, only a small window is displayed on the desktop.
![dfd](https://user-images.githubusercontent.com/56356083/95586180-993c9080-0a40-11eb-8aed-9898731be5cc.jpg)

**Comment by mart0258** [comment]
Okay, I've encountered an unusual issue.  

If borderless is implemented, might need to do an explicit test for _Wing Commander: Privateer_. The method I used to hotwire Dosbox-ECE somehow started crashing without a visible reason, and continued crashing after restarting Dosbox-ECE (until I ran an unmodified version).  This only seemed to react to that specific game, as opposed to other ones.
